src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java

```diff
@@ -347,4 +297,61 @@ private static Object getKeyAsObject(String key, Map<Object, Object> sectionCont
         return null;
     }
 
+	/**
+	 * Writes the current value with the provided trailing key to the provided writer.
+	 *
+	 * @param parserConfig   The parser configuration to use for writing the YAML value.
+	 * @param bufferedWriter The writer to write the value to.
+	 * @param indents        The string representation of the indentation.
+	 * @param trailingKey    The trailing key for the YAML value.
+	 * @param currentValue   The current value to write as YAML.
+	 * @throws IOException If an I/O error occurs while writing the YAML value.
+	 */
+	private static void writeYamlValue(final FileConfiguration parserConfig, final BufferedWriter bufferedWriter, final String indents, final String trailingKey, final Object currentValue) throws IOException {
+		parserConfig.set(trailingKey, currentValue);
+		String yaml = parserConfig.saveToString();
+		yaml = yaml.substring(0, yaml.length() - 1).replace("\n", "\n" + indents);
+		final String toWrite = indents + yaml + "\n";
+		parserConfig.set(trailingKey, null);
+		bufferedWriter.write(toWrite);
+	}
+
+    /**
+     * Writes the value associated with the ignored section to the provided writer,
+     * if it exists in the ignoredSectionsValues map.
+     *
+     * @param ignoredSectionsValues The map containing the ignored section-value mappings.
+     * @param bufferedWriter        The writer to write the value to.
+     * @param fullKey               The full key to search for in the ignoredSectionsValues map.
+     * @throws IOException If an I/O error occurs while writing the value.
+     */
+    private static boolean writeIgnoredSectionValueIfExists(final Map<String, String> ignoredSectionsValues, final BufferedWriter bufferedWriter, final String fullKey) throws IOException {
+        for (final Map.Entry<String, String> entry : ignoredSectionsValues.entrySet()) {
+            if (entry.getKey().equals(fullKey)) {
+                bufferedWriter.write(ignoredSectionsValues.get(fullKey) + "\n");
```

Comment:
Enhance this by moving the if logic before the loop. Loop is only necessary if the fullKey is a sub key, so the else if logic can stay in the loop.

Reply:
Alright can move around the logic, I only keep /preserve your logic. But I see that could be improved.